Sources/Sentry/SentryEvent.m

```diff
@@ -139,6 +140,10 @@ - (void)addSimpleProperties:(NSMutableDictionary *)serializedData
 
     [serializedData setValue:[self serializeBreadcrumbs] forKey:@"breadcrumbs"];
 
+    if (self.serializedBreadcrumbs.count > 0) {
```

Comment:
We're storing the serialized breadcrumbs to disk in JSON, and when we're reading them back (in the case of OOM), we end up with the serialized breadcrumbs again. Instead of turning those dictionaries into proper SentryBreadcrumb instances which then need to be serialized again in SentryEvent, I decided to add a private `serializedBreadcrumbs` property to set the pre-serialized breadcrumbs. It saves a LOT of work - especially turning an NSDictionary into a SentryBreadcrumb is silly when we're just going to do the opposite right after.

Reply:
I think you should merge `serializedBreadcrumbs` with the result from `[self serializeBreadcrumbs]`

Reply:
The only time we set `serializedBreadcrumbs` is when we explicitly set `breadcrumbs` to an empty array. So merging it won't change anything. We really do want to override the breadcrumbs with the stored previous breadcrumbs, so I think the current code makes that very obvious. What do you think?

Reply:
> The only time we set serializedBreadcrumbs is when we explicitly set breadcrumbs to an empty array 

This is true for now, but we should think ahead. Maybe in a future PR we need to set serialized breadcrumbs along side with some SentryBreadcrumbs. 

If there is no bad side to merging both properties I would say we just do it.

Reply:
Fair enough, done

Reply:
`m`: Please add the info you commented above as a code comment.